packages/context/test/unit/manifest.test.ts

```diff
@@ -0,0 +1,466 @@
+import * as fsExtra from "fs-extra";
+import { join, dirname } from "path";
+import {
+  findManifestPath,
+  getCustomViewId,
+  getMainService,
+  getManifestDetails,
+  getServicePath,
+  getUI5Manifest,
+  initializeManifestData,
+} from "../../src/manifest";
+import {
+  Config,
+  ProjectName,
+  ProjectType,
+  TestFramework,
+} from "@ui5-language-assistant/test-framework";
+import { cache } from "../../src/cache";
+import { FileName } from "@sap-ux/project-access";
+import { getProjectData } from "./utils";
+import { URI } from "vscode-uri";
+import { toPosixPath } from "../../src/utils/fileUtils";
+
+const getAppRoot = (projectRoot: string) =>
+  join(projectRoot, "app", "manage_travels", "webapp");
+
+describe("manifest", () => {
+  let testFramework: TestFramework;
+  beforeAll(function () {
+    const useConfig: Config = {
+      projectInfo: {
+        name: ProjectName.cap,
+        type: ProjectType.CAP,
+        npmInstall: true,
+      },
+    };
+    testFramework = new TestFramework(useConfig);
+  }, 5 * 60000 + 10000); // 5 min for initial npm install + 10 sec
```

Comment:
Npm install is launched only when node modules are absent. Otherwise it skipped. The timeout here is just in case if the test  unit is run manually alone.